pkg/pulumiyaml/run.go

```diff
@@ -276,21 +276,48 @@ func (r *runner) setDefaultProviders() error {
 	return nil
 }
 
-// RunTemplate runs the evaluator against a template using the given request/settings.
-func RunTemplate(ctx *pulumi.Context, t *ast.TemplateDecl, loader PackageLoader) error {
-	r := newRunner(ctx, t, loader)
+// PrepareTemplate prepares a template for converting or running
+func PrepareTemplate(ctx *pulumi.Context, t *ast.TemplateDecl, loader PackageLoader) (*runner, *syntax.Diagnostics, error) {
+	// set up context-free runner
+	r := newRunner(t, loader)
 
-	err := r.setDefaultProviders()
+	cwd, err := os.Getwd()
 	if err != nil {
-		return err
+		r.sdiags.Extend(syntax.Error(nil, err.Error(), ""))
+		return nil, nil, err
+	}
+	r.variables[PulumiVarName] = map[string]interface{}{
```

Comment:
@iwahbe I haven't moved the runtime state to `programEvaluator` for now, so we can easily initialize a `programEvaluator` with just a ctx (line 639)